cranelift/assembler-x64/meta/src/generate.rs

```diff
@@ -102,24 +102,18 @@ fn generate_inst_encode_impl(f: &mut Formatter, insts: &[dsl::Inst]) {
     fmtln!(f, "}}");
 }
 
-/// `impl Inst { fn visit... }`
-fn generate_inst_visit_impl(f: &mut Formatter, insts: &[dsl::Inst]) {
-    fmtln!(f, "impl<R: Registers> Inst<R> {{");
-    f.indent(|f| {
-        fmtln!(f, "pub fn visit(&mut self, v: &mut impl RegisterVisitor<R>) {{");
-        f.indent(|f| {
-            fmtln!(f, "match self {{");
-            f.indent_push();
-            for inst in insts {
-                let variant_name = inst.name();
-                fmtln!(f, "Self::{variant_name}(i) => i.visit(v),");
-            }
-            f.indent_pop();
-            fmtln!(f, "}}");
-        });
-        fmtln!(f, "}}");
+/// `impl Inst { fn operands... }`
+fn generate_inst_operands_impl(f: &mut Formatter, insts: &[dsl::Inst]) {
+    f.add_block("impl<R: Registers> Inst<R>", |f| {
+        f.add_block("pub fn operands<'a>(&'a mut self) -> Vec<Operand<'a, R>>", |f| {
```

Comment:
We _could_ use a `smallvec` with a threshold of 4 and be a bit more efficient but I thought, for now, this was fine.